### PR TITLE
Less namespace pollution

### PR DESCRIPTION
**Pull request by erlingrj**
This is still a draft, but wanted some feedback. Essentially, I didn't find a way to remove `windows.h` and friends from `platform.h` since we need it to typdef `lf_mutex_t` etc. I think the only real possibility is to only allow using pointers to lf_thread_t, lf_mutex_t and lf_cond_t in the runtime. And those are just void pointers. They can then be passed into platform files and re-cast. 

I didn't particularly like that idea, so I made a change to avoid including `platform.h` into `lf_types.h` this way, at least the user wont get their namespace polluted unless they manually include `platform.h`

### CODE REVIEW

File: core/platform/lf_C11_threads_support.c
Comment (edwardalee) [inline core/platform/lf_C11_threads_support.c:1]: Should this file have function comments? Shouldn't they just be in  `platform.h`? Comments here probably should only be about the particular customization to C11.  Same question applies to all the .c files.

File: core/platform/lf_C11_threads_support.c [hunk 1]
```diff
@@ -0,0 +1,123 @@
+#include "platform.h"
+#include <threads.h>
+#include <stdlib.h>
+#include <stdint.h> // For fixed-width integral types
+
+
+/**
+ * Create a new thread, starting with execution of lf_thread
+ * getting passed arguments. The new handle is tored in thread.
```
Comment (edwardalee) [inline core/platform/lf_C11_threads_support.c:9]:
```suggestion
 * getting passed arguments. The new handle is stored in the thread argument.
```

File: core/platform/lf_C11_threads_support.c [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+ *
+ * @return 0 on success, error number otherwise (see thrd_create()).
+ */
+int lf_thread_create(lf_thread_t* thread, void *(*lf_thread) (void *), void* arguments) {
+    return thrd_create((thrd_t*)thread, (thrd_start_t)lf_thread, arguments);
+}
+
+/**
+ * Make calling thread wait for termination of the thread.  The
+ * exit status of the thread is stored in thread_return, if thread_return
```
Comment (edwardalee) [inline core/platform/lf_C11_threads_support.c:19]:
```suggestion
 * exit status of the thread is stored in thread_return if thread_return
```

File: core/platform/lf_C11_threads_support.c [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    switch (return_value) {
+        case thrd_timedout:
+            return_value = LF_TIMEOUT;
+            break;
+
+        default:
+            break;
+    }
+    return return_value;
+}
```
Comment (edwardalee) [inline core/platform/lf_C11_threads_support.c:123]:
```suggestion
}

```

File: core/platform/lf_POSIX_threads_support.c [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    switch (return_value) {
+        case ETIMEDOUT:
+            return_value = LF_TIMEOUT;
+            break;
+
+        default:
+            break;
+    }
+    return return_value;
+}
```
Comment (edwardalee) [inline core/platform/lf_POSIX_threads_support.c:146]:
```suggestion
}

```

**Review by edwardalee — APPROVED** [review]
This looks like a pretty good solution to me! Minor suggestions included.

**Comment by petervdonovan** [comment]
Overall this approach is the one that I would most prefer and I think we should merge it when the tests pass.

I agree with Edward's observation about duplicated comments.

File: core/platform/lf_C11_threads_support.c
Comment (erlingrj) [reply core/platform/lf_C11_threads_support.c:1]: Thanks, have addressed this now

**Review by petervdonovan — APPROVED** [review]
Thanks Erling!

**Review by edwardalee — APPROVED** [review]
LGTM!

**Pull request merged**